### ISSUE TITLE
Error on progress graph for Habit or Daily with null history

### ISSUE THREAD

**Issue by SabreCat** [HabitRPG/habitica#3691]
In the past, the progress graph icon didn't appear on a Habit until you'd clicked it at least once. Now, the icon appears immediately when you create a new Habit, and when you click it, you get this error:

![screenclip](https://cloud.githubusercontent.com/assets/4501321/3503390/f0893008-062f-11e4-8619-357ef0a49897.png)

**Comment by SabreCat** [comment]
Also affects Dailies. Updated title.

**Comment by tristan-hs** [comment]
Looks like any progress button with no record in the associated table is getting this. I also found it on the experience bar on my first day, the to-do progress button, and any new daily or habit.

**Comment by Linskeyd** [comment]
The code below is inside of rootCtrl.js

```
$rootScope.toggleChart = function(id, task) {
      var history = [], matrix, data, chart, options;
      switch (id) {
        case 'exp':
          $rootScope.charts.exp = !$rootScope.charts.exp;
          history = User.user.history.exp;
          break;
        case 'todos':
          $rootScope.charts.todos = !$rootScope.charts.todos;
          history = User.user.history.todos;
          break;
        default:
          $rootScope.charts[id] = !$rootScope.charts[id];
          history = task.history;
          if (task && task._editing) task._editing = false;
      }
      matrix = [['Date', 'Score']];
      _.each(history, function(obj) {
        matrix.push([moment(obj.date).format('MM/DD/YY'), obj.value]);
      });
      data = google.visualization.arrayToDataTable(matrix);
      options = {
        title: window.env.t('history'),
        backgroundColor: {
          fill: 'transparent'
        },
        hAxis: {slantedText:true, slantedTextAngle: 90},
        height:270,
        width:300
      };
      chart = new google.visualization.LineChart($("." + id + "-chart")[0]);
      chart.draw(data, options);
    };
```

I went and screwed around on the visualization playground here https://code.google.com/apis/ajax/playground/?type=visualization#line_chart and deleted all the data except for the headers to simulate the _.each statement from underscore not being able to iterate over any data.  If history comes back null from User.user.history for exp or todos, then it needs to display a message saying that there is no history found, especially in the edge case of exp being 0 at level 1 when a user first starts and has no history.  Seems somewhere though that a problem exists with the way the exp/todos are saved/retrieved.  

It looks like history is given a default in underscore below in 20131028_task_subdocs_tags_invites.js

```
// Fix missing history
  // -------------------------
  _.defaults(user, {history:{}});
  _.defaults(user.history,{exp:[], todos:[]});
```

I'm not sure where the history is going in and being appended to the user model.

**Comment by deilann** [comment]
Actually, I think the solution is to go back to not displaying the progress button if there's no progress.

**Comment by Linskeyd** [comment]
That would mean you wouldn't be able to lazy check the history anymore.

**Comment by deilann** [comment]
I'm not sure what you mean. In the past the progress bar only showed up
once there was progress to show. If we fix it so it goes back like that,
this bug doesn't exist.

**Comment by jklein** [comment]
Looks like this issue can be closed? It's no longer a bug since #3818 was merged.
